### PR TITLE
Updated methods and vars

### PR DESCRIPTION
**Pull request by AshishSynoriq**
update sdk version and methods

### CODE REVIEW

File: android/src/main/kotlin/com/hypertrack/sdk/flutter/HyperTrackPlugin.kt [hunk 1]
```diff
@@ -82,25 +88,34 @@ public class HyperTrackPlugin(): FlutterPlugin, MethodCallHandler, StreamHandler
...
       return
     }
-
+    
     when (call.method) {
       "getDeviceId" -> result.success(sdk.deviceID)
       "isRunning" -> result.success(sdk.isRunning)
+      "isTracking" -> result.success(sdk.isTracking)
+      "getAvailability" -> result.success(sdk.availability.toString())
+      "setAvailability" -> setAvailability(result,sdk,call.arguments<Boolean>()?:false)
```
Comment (ferologics) [inline android/src/main/kotlin/com/hypertrack/sdk/flutter/HyperTrackPlugin.kt:97]: should this default to `false`?

File: android/src/main/kotlin/com/hypertrack/sdk/flutter/HyperTrackPlugin.kt [hunk 2]
```diff
@@ -167,34 +182,48 @@ public class HyperTrackPlugin(): FlutterPlugin, MethodCallHandler, StreamHandler
...
+  private fun setAvailability(result: MethodChannel.Result, sdk: HyperTrack, arguments: Boolean) {
+    Log.d(TAG, "setAvailability called")
+  if (arguments == true) {
+    sdk.setAvailability(Availability.AVAILABLE)
+     }
+  else if (arguments == false) {
+    sdk.setAvailability(Availability.UNAVAILABLE)
+    result.success(null)
+     }
+  }
```
Comment (ferologics) [inline android/src/main/kotlin/com/hypertrack/sdk/flutter/HyperTrackPlugin.kt:208]: this whole method has strange formatting, pls adjust

File: android/src/main/kotlin/com/hypertrack/sdk/flutter/HyperTrackPlugin.kt [hunk 3]
```diff
@@ -167,34 +182,48 @@ public class HyperTrackPlugin(): FlutterPlugin, MethodCallHandler, StreamHandler
...
+  
+
+
```
Comment (ferologics) [inline android/src/main/kotlin/com/hypertrack/sdk/flutter/HyperTrackPlugin.kt:211]:
```suggestion
```
remove extraneous empty newlines

File: android/src/main/kotlin/com/hypertrack/sdk/flutter/HyperTrackPlugin.kt [hunk 4]
```diff
@@ -167,34 +182,48 @@ public class HyperTrackPlugin(): FlutterPlugin, MethodCallHandler, StreamHandler
...
 
-  private fun syncDeviceSettings(result : Result, sdk : HyperTrack) {
+  private fun syncDeviceSettings(result: MethodChannel.Result, sdk : HyperTrack) {
     Log.d(TAG, "syncDeviceSettings called")
 
     sdk.syncDeviceSettings()
     result.success(null)
   }
 
+
```
Comment (ferologics) [inline android/src/main/kotlin/com/hypertrack/sdk/flutter/HyperTrackPlugin.kt:226]:
```suggestion
```
remove empty newline

**Review by ferologics — CHANGES_REQUESTED** [review]
mostly nitpicks, but looks good overall. can you provide instructions about testing the new methods?

File: android/src/main/kotlin/com/hypertrack/sdk/flutter/HyperTrackPlugin.kt (hunk 1 above)
Comment (tulushev) [reply android/src/main/kotlin/com/hypertrack/sdk/flutter/HyperTrackPlugin.kt:97]: @AshishSynoriq  There should be no defaults. `setAvailablility` on Dart should accept a non-null value and pass it here.

File: lib/hypertrack_platform_interface.dart [hunk 5]
```diff
@@ -87,25 +121,25 @@ abstract class HypertrackPlatformInterface extends PlatformInterface {
...
   syncDeviceSettings() => _methodChannel.invokeMethod('syncDeviceSettings');
 
-  Stream<bool> get isRunningStatus {
-    Stream<bool>? a = _eventChannel
+  Stream<bool> get isAvailable {
+    return _availabilitySubscription
         .receiveBroadcastStream()
-        .map((dynamic event) => event == 'start');
-    return a;
+        .where((event) => event is bool)
```
[the remark this answers is not in the document]
Comment (AshishSynoriq) [reply lib/hypertrack_platform_interface.dart:135]: The event stream sends updates for both availability and the tracking state in the receiveBroadcastStream method, so to distinguish this is used. Thats why I have not implemented the getAvailability on iOS yet, as I believe this issue can be prevented there.

**Pull request closed without merging**